viewer/packages/cad-model/src/picking/PickingHandler.ts

```diff
@@ -92,16 +92,24 @@ export class PickingHandler {
   ): Promise<IntersectCadNodesResult[]> {
     const results: IntersectCadNodesResult[] = [];
     const release = await this._mutex.acquire();
+    const cadNodesVisibility = cadNodes.map(p => p.visible);
```

Comment:
I think we can avoid this one and only use the `visibleCadNodes` array. In the end, we can just set all of the elements in that array to visible :+1: